app/controllers/api/v2/preupgrade_reports_controller.rb

```diff
@@ -3,17 +3,41 @@
 module Api
   module V2
     class PreupgradeReportsController < ::Api::V2::BaseController
-      before_action :find_resource, only: %i[show]
+      include ApiAuthorizer
 
       api :GET, '/preupgrade_reports/', N_('List Preupgrade reports')
       param_group :search_and_pagination, ::Api::V2::BaseController
       def index
-        @preupgrade_reports = resource_scope_for_index
+        @preupgrade_reports = PreupgradeReport.search_for(*search_options)
```

Comment:
`resource_scope_for_index` sets a couple of instance variables which are then used when rendering the response

```
$ curl -u Foo:changeme http://localhost:3000/api/v2/preupgrade_reports/ | jq                                                
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                                                                                                                                             
                                 Dload  Upload   Total   Spent    Left  Speed
100  3146    0  3146    0     0   9987      0 --:--:-- --:--:-- --:--:--  9987
{                    
  "total": 0,                    
  "subtotal": 0,     
  "page": 1,                  
  "per_page": 20,                             
  "search": null,                            
  "sort": {                                    
    "by": null,    
    "order": null                              
  },                                                                                                                                                          
  "results": [                                                                 
    {                                                                        
      "id": 32,                                                               
      "host_id": 8,           
      "reported_at": "2020-05-04 14:28:07 UTC",
      "status": 0,           
      "metrics": {},                           
      "type": "PreupgradeReport",
      "origin": null,                          
      "job_invocation_id": 158   
    },
    ...
  }
]

I believe that not using this (and therefore not setting the variables) is the reason why `total` shows up as 0, even though there are some results
```